src/realm/alloc_slab.cpp

```diff
@@ -235,7 +235,8 @@ MemRef SlabAlloc::do_alloc(size_t size)
 #endif
 
     char* addr = reinterpret_cast<char*>(entry);
-    REALM_ASSERT_EX(addr == translate_in_slab(ref), addr, ref, get_file_path_for_assertions());
+    // REALM_ASSERT_EX(addr == translate_in_slab(ref), addr, ref, get_file_path_for_assertions());
+    REALM_ASSERT_EX(addr == translate(ref), addr, ref, get_file_path_for_assertions());
```

Comment:
is this actually correct? can we be sure that the memory is on slab at this stage?